helper/common/shared_state.go

```diff
@@ -0,0 +1,31 @@
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// Used to set variables which we need to access later in the build, where
+// state bag and config information won't work
+func sharedStateFilename(suffix string) string {
+	uuid := os.Getenv("PACKER_RUN_UUID")
+	return filepath.Join(os.TempDir(), fmt.Sprintf("packer-%s-%s", uuid, suffix))
+}
+
+func SetSharedState(key string, value string) error {
+	return ioutil.WriteFile(sharedStateFilename(key), []byte(value), 0644)
```

Comment:
should this maybe be 0600 now that we're possibly writing passwords?

Reply:
good idea.